### PR TITLE
[Bitcore-Client] Adds a method for bumping fee for stuck TX

### PR DESCRIPTION
**Pull request by escottalexander**
This PR works for EVM based chains only. UTXO support will be added in another PR.

Tested against my local devnet and Goerli. Works with a raw tx or a tx id.

### CODE REVIEW

**Review by kajoseph — CHANGES_REQUESTED** [review]
A few minor things. I haven't tested it yet

File: packages/bitcore-client/bin/wallet-bump-tx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  .version(require('../package.json').version)
+  .option('--name <name>', 'REQUIRED - Wallet Name')
+  .option('--txid [txid]', 'REQUIRED (or provide raw tx) - string transaction id')
+  .option('--rawtx [rawtx]', 'REQUIRED (or provide tx id) - string raw transaction')
+  .option(
+    '--gwei [gwei]',
+    'optional  (defaults to current market rate) for use with EVM based chains - number gwei (in eth/matic)'
+  )
+  .option('--path [path]', 'optional - Custom wallet storage path')
+  .option('--storageType [storageType]', 'optional - name of the database to use (default Level)')
```
[the remark this answers is not in the document]
Comment (escottalexander) [reply packages/bitcore-client/bin/wallet-bump-tx:20]: I assume you mean storageType. I can change it but all the other utils have it like this. If I change it here, I should change them all.

File: packages/bitcore-client/bin/wallet-bump-tx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    } else if (rawtx) {
+      existingTx = ethers.utils.parseTransaction(rawtx);
+    } else {
+      throw new Error('Must provide either a raw transaction or transaction id.');
+    }
+
+    const existingGasPriceInWei = Number(existingTx.gasPrice);
+    let newFeeInWei;
+    if (gwei) {
+      newFeeInWei = Number(gwei) * 1000000000;
```
Comment (kajoseph) [inline packages/bitcore-client/bin/wallet-bump-tx:42]: I would suggest making 1000000000 a variable and using it everywhere we have `1000000000`. Things like this make it easy to miss a zero in one place or something

**Review by kajoseph — CHANGES_REQUESTED** [review]
We should do a check and stop if the new gwei <= old gwei. If I don't specify --gwei at all, then it displays `Bumping gas price from 20 GWei to 20 GWei` and asks me if I want to broadcast, but it's easy to miss that message

File: packages/bitcore-client/bin/wallet-bump-tx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kajoseph) [reply packages/bitcore-client/bin/wallet-bump-tx:20]: I was actually referring to the description. --path starts with "Custom wallet..." whereas --storageType starts with "name of...".

Also, we have REQUIRED in all upper case and optional in all lower case...which I suppose is fine but not having REQUIRED implies it's optional so we really don't need "optional" for the optional flags.

[the remark this answers is not in the document]
Comment (kajoseph) [reply packages/bitcore-client/bin/wallet-bump-tx:20]: Also, --gwei starts with the default whereas --storageType ends with the default. I imagine you're mostly copying and pasting from other utils, but I'd like to increase our standard for attention to detail

**Comment by escottalexander** [comment]
> We should do a check and stop if the new gwei <= old gwei. If I don't specify --gwei at all, then it displays Bumping gas price from 20 GWei to 20 GWei and asks me if I want to broadcast, but it's easy to miss that message

@kajoseph At one point I actually had the check you recommend but as I thought about it it seemed like it isn't necessary. If you provide --gwei then it checks to make sure it is more than the existing gwei. If you don't specify the gwei then it gets the current market rate which should always be greater than the tx's gwei granted it's truly stuck or else the original will go through momentarily and this util will error for using the same nonce. If you think it is still worthwhile for running against devnet then I can add it but I almost think it's more useful for testing against devnet to not handle this edge case.

**Comment by kajoseph** [comment]
@escottalexander that seems sensible. We can leave it as is, then

**Comment by escottalexander** [comment]
@kajoseph Thanks for the feedback. The requested changes have been committed.
